drivers/interrupt_controller/intc_intel_vtd.c

```diff
@@ -109,6 +116,8 @@ static int vtd_ictl_init(const struct device *dev)
 	vtd_send_cmd(dev, VTD_GCMD_SIRTP, VTD_GSTS_SIRTPS);
 	vtd_send_cmd(dev, VTD_GCMD_IRE, VTD_GSTS_IRES);
 
+	/* enable compactability mode */
+	vtd_send_cmd(dev, VTD_GCMD_CFI, VTD_GSTS_CFIS);
```

Comment:
CFIS must not be enabled if X2APIC mode is enabled (which is the default). That's just adding a bug here. Again check the VT-D specs.

Reply:
The compactability mode is not enabled then Compatibility format interrupts are blocked. Please read "Interrupt-Remapping Hardware Operation" section carefully. If not enabled then current legacy interrupt will block (you are not enabled remap for them).

Reply:
See https://github.com/zephyrproject-rtos/zephyr/pull/36731/commits/0d1eb9fa8f2892f3a48a3c2c51d2c98908b9becf
and re-read the specs section 5.1.4. It's only valid for xapic mode, not x2apic, and only if you don't want to remap ioapic generated interrupts (which are in compat format by default). See https://github.com/zephyrproject-rtos/zephyr/pull/36731/commits/4fc22839bff7e70e5ae4d8d7f497d432adb08c61